### PR TITLE
fix: exclude paths from codeql evaluation, skip e2e tests on scheduled runs and update codeowners

### PR DESCRIPTION
**Pull request by ahmedmuhsin**
- add exclude paths for codeql
- prevent end-to-end tests from running on scheduled codeql runs

### Issue describing the changes in this PR

resolves #issue_for_this_pr

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes are added to the `CHANGELOG.md`
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: eng/ci/official-build.yml
Comment (manvkaur) [inline eng/ci/official-build.yml:19]: Why do we want to run e2e tests on official build for times other than scheduled builds?
It is probably good to run them after functional or code changes merged to main to see if the break some e2e flow.

**Review by manvkaur — APPROVED** [review]

**Review by manvkaur — DISMISSED** [review]

**Comment by manvkaur** [comment]
Let's also update the CodeOwners, right now review from Mamoun is mandatory.

File: eng/ci/official-build.yml
Comment (ahmedmuhsin) [reply eng/ci/official-build.yml:19]: This flag is true by default however, for scheduled runs, like codeql, the option is forced to false. So it will run when there are code changes (unless the user explicitly turns the flag off).

File: CODEOWNERS
Comment (manvkaur) [inline CODEOWNERS]: should we use this or `@Azure/azure-functions-java-worker-admins `

File: CODEOWNERS
Comment (manvkaur) [inline CODEOWNERS]: is the leading space causing problem?
It seems only Mamoun's alias was added and not Pragna's previously because there is a leading space in front of her alias.

- Check same for library repo too.
Reference of a working one from bundles repo - https://github.com/Azure/azure-functions-extension-bundles/blob/main/CODEOWNERS

Comment (ahmedmuhsin) [reply CODEOWNERS]: I've updated it for both repos. How do you check if it worked?

Comment (manvkaur) [reply CODEOWNERS]: Looks like it is not working, 
image
Hovering over the error message suggests that the team is not being found or not visible publicly

Comment (manvkaur) [reply CODEOWNERS]: Updated to correct role

**Review by manvkaur — APPROVED** [review]

**Pull request merged**
